plugins/aea-ledger-hwi/aea_ledger_hwi/hwi.py

```diff
@@ -0,0 +1,338 @@
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""Ethereum module wrapping the public and private key cryptography and ledger api."""
+
+import logging
+import warnings
+from typing import Any, Callable, Dict, List, Optional, Union, cast
+
+from aea_ledger_ethereum.ethereum import (
+    EthereumApi,
+    EthereumFaucetApi,
+    EthereumHelper,
+    TESTNET_NAME,
+    set_wrapper_for_web3py_session_cache,
+)
+from aea_ledger_hwi.account import HWIAccount
+from eth_account.datastructures import HexBytes, SignedTransaction
+from eth_account.messages import encode_defunct
+from eth_account.signers.local import LocalAccount
+from web3.datastructures import AttributeDict
+from web3.types import TxData, TxReceipt
+
+from aea.common import JSONLike
+from aea.crypto.base import Crypto
+
+
+_default_logger = logging.getLogger(__name__)
+
+_ETHEREUM_HWI = "ethereum_hwi"
+DEFAULT_DEVICE_INDEX = 0
+DEFAULT_KEYPAIR_INDEX = 0
+
+
+class SignedTransactionTranslator:
+    """Translator for SignedTransaction."""
+
+    @staticmethod
+    def to_dict(signed_transaction: SignedTransaction) -> Dict[str, Union[str, int]]:
+        """Write SignedTransaction to dict."""
+        signed_transaction_dict: Dict[str, Union[str, int]] = {
+            "raw_transaction": cast(str, signed_transaction.rawTransaction.hex()),
+            "hash": cast(str, signed_transaction.hash.hex()),
+            "r": cast(int, signed_transaction.r),
+            "s": cast(int, signed_transaction.s),
+            "v": cast(int, signed_transaction.v),
+        }
+        return signed_transaction_dict
+
+    @staticmethod
+    def from_dict(signed_transaction_dict: JSONLike) -> SignedTransaction:
+        """Get SignedTransaction from dict."""
+        if (
+            not isinstance(signed_transaction_dict, dict)
+            and len(signed_transaction_dict) == 5
+        ):
+            raise ValueError(  # pragma: nocover
+                f"Invalid for conversion. Found object: {signed_transaction_dict}."
+            )
+        signed_transaction = SignedTransaction(
+            rawTransaction=HexBytes(
+                cast(str, signed_transaction_dict["raw_transaction"])
+            ),
+            hash=HexBytes(cast(str, signed_transaction_dict["hash"])),
+            r=cast(int, signed_transaction_dict["r"]),
+            s=cast(int, signed_transaction_dict["s"]),
+            v=cast(int, signed_transaction_dict["v"]),
+        )
+        return signed_transaction
+
+
+class AttributeDictTranslator:
```

Comment:
Can we not use or at least extend the one from base (`open-aea-ledger-ethereum`)?